linode_api4/linode_client.py

```diff
@@ -402,6 +402,133 @@ def __call__(self):
+
+    def user_preferences(self):
+        """
+        View a list of user preferences tied to the OAuth client that generated the token making the request.
+        """
+
+        result = self.client.get(
+            "{}/preferences".format(Profile.api_endpoint), model=self
+        )
+
+        return MappedObject(**result)
+
+    def security_questions(self):
+        """
+        Returns a collection of security questions and their responses, if any, for your User Profile.
+        """
+
+        result = self.client.get(
+            "{}/security-questions".format(Profile.api_endpoint), model=self
+        )
+
+        return MappedObject(**result)
+
+    def security_questions_answer(self, questions):
+        """
+        Adds security question responses for your User. Requires exactly three unique questions.
+        Previous responses are overwritten if answered or reset to null if unanswered.
+
+        Example question:
+        {
+            "question_id": 11,
+            "response": "secret answer 3"
+        }
+        """
+
+        if len(questions) != 3:
+            raise ValueError("Exactly 3 security questions are required.")
+
+        params = {"security_questions": questions}
+
+        result = self.client.post(
+            "{}/security-questions".format(Profile.api_endpoint),
+            model=self,
+            data=params,
+        )
+
+        return MappedObject(**result)
+
+    def user_preferences_update(self, **preferences):
+        """
+        Updates a user’s preferences.
+        """
+
+        result = self.client.put(
+            "{}/preferences".format(Profile.api_endpoint),
+            model=self,
+            data=preferences,
+        )
+
+        return MappedObject(**result)
+
+    def phone_number_delete(self):
+        """
+        Delete the verified phone number for the User making this request.
+        """
+
+        self.client.delete(
+            "{}/phone-number".format(Profile.api_endpoint), model=self
+        )
+
+    def phone_number_verify(self, otp_code):
+        """
+        Verify a phone number by confirming the one-time code received via SMS message
+        after accessing the Phone Verification Code Send (POST /profile/phone-number) command.
+        """
+
+        if not otp_code:
+            raise ValueError("OTP Code required to verify phone number.")
+
+        params = {"otp_code": otp_code}
+
+        self.client.post(
+            "{}/phone-number/verify".format(Profile.api_endpoint),
+            model=self,
+            data=params,
+        )
+
+    def phone_number_verification_code_send(self, iso_code, phone_number):
+        """
+        Send a one-time verification code via SMS message to the submitted phone number.
+        """
+
+        if not iso_code:
+            raise ValueError("ISO Code required to send verification code.")
+
+        if not phone_number:
+            raise ValueError("Phone Number required to send verification code.")
```

Comment:
Since the parameters are not optional parameters, we can generally consider them as required parameters and wouldn't need manually checking on them. I think it's also okay to keep it, but I would recommend not to check existence of a required parameter in the future. You can have some comprehensive check and validation, for example, validating phone number, of course.

Nothing in this comment need to be addressed for this PR, just some thoughts.

Reply:
Thanks for the tip! My thought was to make sure that the user didn't pass `None` explicitly since the parameters are required for the API call.